paddle/math/Matrix.cpp

```diff
@@ -371,7 +383,7 @@ MatrixPtr GpuMatrix::getTranspose() {
   }
 }
 
-void GpuMatrix::transpose(MatrixPtr matTrans, bool memAlloc) {
+void GpuMatrix::transpose(MatrixPtr& matTrans, bool memAlloc) {
   if (memAlloc) {
```

Comment:
Yes. When memAlloc is true, the pointer matTrans will change, and thus the reference should be passed so that outside the function the pointer is also changed.